### ISSUE TITLE
Dashboards: Animation lost on HighchartsComponent on polling

### ISSUE THREAD

**Issue by TorsteinHonsi** [highcharts/highcharts#20483]
#### Expected behaviour
Charts should animate smoothly like [they do with polling in the data module](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/data/livedata-rows/). Only one chart redraw should be performed.

#### Actual behaviour
Animation is lost. On each iteration, the chart is redrawn 4 times, the first one without data points, causing animation to be lost, and also a waste of CPU due to excess redrawing.

#### Live demo with steps to reproduce
https://jsfiddle.net/highcharts/vu3fny4h/
Look at the console. When we start polling, we immediately have two redraws. Those are not the focus of this issue.

Then when the first polling cycle runs, we have four calls to `redraw`.

#### Product version
v1.2.1

#### Debugging results
It looks like the DataTable is either too eager to spark events, or the polling mechanism should not pick them all up.
* For each polling cycle, `DataTable.deleteRows` runs, emitting an event that ultimately redraws all connected components with empty data. That should not happen, because we know that new data is coming in.
* `DataTable.setModifier` also sparks events that redraw the components through the `afterSetModifier` event.

When I comment out these two event emitters, the chart only redraws once, and the animation works as intended:
```diff
diff --git a/ts/Data/DataTable.ts b/ts/Data/DataTable.ts
index 704bd38b55..b9d7522b15 100644
--- a/ts/Data/DataTable.ts
+++ b/ts/Data/DataTable.ts
@@ -502,6 +502,7 @@ class DataTable implements DataEvent.Emitter {
             );
         }
 
+        /*
         table.emit({
             type: 'afterDeleteRows',
             detail: eventDetail,
@@ -509,6 +510,7 @@ class DataTable implements DataEvent.Emitter {
             rowIndex: (rowIndex || 0),
             rows: deletedRows
         });
+        */
 
         return deletedRows;
     }
@@ -1459,12 +1461,14 @@ class DataTable implements DataEvent.Emitter {
 
         return promise
             .then((table): this => {
+                /*
                 table.emit({
                     type: 'afterSetModifier',
                     detail: eventDetail,
                     modifier,
                     modified: table.modified
                 });
+                */
                 return table;
             })['catch']((error): this => {
                 table.emit({
```

In summary, we should economize with the events causing chart redraws. Either we shouldn't emit events in some cases, or the handlers should be adjusted to cancel them. Furthermore, this probably applies to all components, and for those that don't have the animation issue, it will still be a CPU sink.

**Comment by TorsteinHonsi** [comment]
PS: In [some cases](https://github.com/highcharts/highcharts/blob/124efb105adaccd53fc7d6b68d3fc03afff22113/ts/Dashboards/Plugins/HighchartsSyncHandlers.ts#L231) we use a `trigger` property on the event details to keep track of what caused the event thread. Depending on the trigger, we can for example avoid excessive redrawing. Maybe that could be a solution here too.

**Comment by TorsteinHonsi** [comment]
Broken down further: https://jsfiddle.net/highcharts/6wjuvbrL/

* There are two redraws on load. The first is probably legit because the data arrives and populates the chart. The second is a chart resize because the container size changes. Should probably be fixed, but it's another issue.
* Click the button. There are now 3 (or four?) chart redraws. None of these should happen because the data didn't change.

**Comment by sebastianbochan** [comment]
The series is updated (and redraw) when the event `tableChange` is triggered. 

https://github.com/highcharts/highcharts/blob/a91364fd6b8e13490a2285f27e4889a032fc052c/ts/Dashboards/Plugins/HighchartsComponent.ts#L432-L433

The `tableChange` is emitted on all of these actions:

https://github.com/highcharts/highcharts/blob/a91364fd6b8e13490a2285f27e4889a032fc052c/ts/Dashboards/Components/Component.ts#L565-L570

So I think that it should be simplified.

Can we distiquish when data is new in table/dataTable or connector? Maybe it would be enough to add this information in datapool/connector or dataTable?

**Comment by TorsteinHonsi** [comment]
@sebastianbochan @bre1470 

> Can we distiquish when data is new in table/dataTable or connector?

As a start, we see that when polling, the polling mechanism first calls `deleteRows`, which triggers the `alfterDeleteRow` event, which causes a chart redraw. 

In this case, it is safe to say that we should economize, because we already know that the delete is temporary, and there are new rows coming in immediately. Consequently, we should either not fire the event, or fire the event with some flag (`e.trigger`) that ultimately lets us decide that we don't update the component at this stage.

This is a simplification, but if it works out, we can probably apply the same economization with the other redraws, until we are left with the final redraw that provides a smooth animation into the updated data values.

**Comment by sebastianbochan** [comment]
It looks like the last changes with delete columns instead of rows do not have an influence on updates in the case: https://jsfiddle.net/BlackLabel/j6sq7wur/ The values are still added.

FYI @bre1470
